guides/source/deploy_performance_tuning.md

```diff
@@ -0,0 +1,146 @@
+
+In a few cases it may not make sense to preload your application. In that case it's possible to turn off application preloading.
+
+Puma preloads your application by default by calling +preload_app!+ in +config/puma.rb+. If you remove this call, your application will not be preloaded.
+
+Performance Testing
+-------------------
+
+Settings from "Choosing Default Settings" are much better than using the initial Rails defaults. But your specific application may have unusual needs or benefit from different configuration options.
+
+The best way to choose your application's settings is to test the performance of your application with a simulated production workload. This load testing can be a lot of work. You should save your application's load testing code so you can re-run the tests with future versions of your application.
+
+Performance testing is a deep subject. This guide gives only simple guidelines.
+
+### Load Testers
+
+You will need a load testing program to make requests of your application. This can be a dedicated load testing program of some kind, or you can write a small application to make HTTP requests and track how long they take. You should not normally check the time in your Rails logfile. That time is only how long Rails took to process the request. It does not include time taken by the application server.
+
+Sending many simultaneous requests and timing them can be difficult. It is easy to introduce subtle measurement errors. Normally you should use a load testing program, not write your own. Many load testers are simple to use and many excellent load testers are free.
+
+### What to Measure
+
+Throughput is the number of requests per second that your application successfully processes. Any good load testing program will measure it. Throughput is normally a single number for each load test.
+
+Latency is the delay from the time the request is sent until its response is successfully received. Each individual request will have its own latency.
+
+[Percentile](https://en.wikipedia.org/wiki/Percentile_rank) latency gives the latency where a certain percentage of requests have better latency than that. For instance, P90 is the 90th-percentile latency. The P90 is the latency for a single load test where only 10% of requests look longer than that to process. The P50 is the latency such that half your requests were slower, also called the median latency.
+
+"Tail latency" refers to high-percentile latencies. For instance, the P99 is the latency such that only 1% of your requests was worse. P99 is a tail latency. P50 is not a tail latency.
+
+### What You Can Change
+
+You can change the number of processes or threads in your test. You can also change other Puma configuration options such as wait_for_less_busy_worker, though you don't normally need to change them.
```

Comment:
```suggestion
You can change the number of threads in your test.
```

Threads are pretty much the only meaningful knob to turn. Which is good because multi-dimenstional testing is a pain.

Reply:
I've clarified when and how you may want to vary number of processes and mentioned memory testing.